src/sentry/monitors/processing_errors.py

```diff
@@ -36,3 +54,127 @@ class ProcessingErrorType(Enum):
+
+    def to_dict(self) -> CheckinProcessingErrorData:
+        return {
+            "errors": [error.to_dict() for error in self.errors],
+            "checkin": self.checkin.to_dict(),
+        }
+
+    @classmethod
+    def from_dict(cls, data: CheckinProcessingErrorData) -> CheckinProcessingError:
+        return cls(
+            errors=[ProcessingError.from_dict(error) for error in data["errors"]],
+            checkin=CheckinItem.from_dict(data["checkin"]),
+        )
+
+
+class CheckinProcessErrorsManager:
+    def _get_cluster(self):
+        return redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
+
+    def store(self, error: CheckinProcessingError, monitor: Monitor | None):
+        if monitor is None:
+            # Attempt to get the monitor from the checkin info if we failed to retrieve it during ingestion
+            try:
+                monitor = Monitor.objects.get(
+                    project_id=error.checkin.message["project_id"],
+                    slug=error.checkin.payload["monitor_slug"],
+                )
+            except Monitor.DoesNotExist:
+                pass
+        if monitor:
+            error_identifier = f"monitor:{monitor.id}"
+        else:
+            error_identifier = f'project:{error.checkin.message["project_id"]}'
+
+        error_key = f"monitors.processing_errors.{error_identifier}"
+        serialized_error = json.dumps(error.to_dict())
+        redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
+        pipeline = redis_client.pipeline(transaction=False)
+        pipeline.zadd(error_key, {serialized_error: error.checkin.ts.timestamp()})
+        # Cap the error list to the `MAX_ERRORS_PER_SET` most recent errors
+        pipeline.zremrangebyrank(error_key, 0, -(MAX_ERRORS_PER_SET + 1))
+        pipeline.expire(error_key, MONITOR_ERRORS_LIFETIME)
+        pipeline.execute()
+
+    def build_monitor_identifier(self, monitor: Monitor) -> str:
+        return f"monitor:{monitor.id}"
+
+    def get_for_monitor(self, monitor: Monitor) -> list[CheckinProcessingError]:
+        return self._get_for_entity(self.build_monitor_identifier(monitor))
+
+    def build_project_identifier(self, project: Project) -> str:
+        return f"project:{project.id}"
+
+    def get_for_project(self, project: Project) -> list[CheckinProcessingError]:
+        return self._get_for_entity(self.build_project_identifier(project))
+
+    def _get_for_entity(self, identifier: str) -> list[CheckinProcessingError]:
+        redis = self._get_cluster()
+        error_key = f"monitors.processing_errors.{identifier}"
+        raw_errors = redis.zrange(error_key, 0, MAX_ERRORS_PER_SET, desc=True)
+        return [CheckinProcessingError.from_dict(json.loads(raw_error)) for raw_error in raw_errors]
+
+
+def handle_processing_errors(item: CheckinItem, error: CheckinValidationError):
+    try:
+        organization = Organization.objects.get(project__id=item.message["project_id"])
```

Comment:
should we get from cache?

Reply:
There's no cache for doing this by project id https://github.com/getsentry/sentry/blob/fa0930966ce70d49060dda79c9f951b334d35929/src/sentry/models/organization.py#L203

I could fetch the project by cache, then fetch the org by cache... maybe that's better